installscripts/jazz-terraform-unix-noinstances/ecs.tf

```diff
@@ -0,0 +1,389 @@
+
+  vars {
+    image           = "${var.codeq_docker_image}"
+    ecs_container_name = "${var.envPrefix}_ecs_container_codeq"
+    log_group       = "${aws_cloudwatch_log_group.ecs_fargates_cwlogs.name}"
+    prefix_name     = "${var.envPrefix}_ecs_task_definition_codeq"
+    region          = "${var.region}"
+    gitlab_passwd    = "${var.cognito_pool_password}"
+  }
+  depends_on = ["aws_cloudwatch_log_group.ecs_fargates_cwlogs"]
+}
+
+resource "aws_ecs_task_definition" "ecs_task_definition" {
+  count = "${var.dockerizedJenkins}"
+  family                   = "${var.envPrefix}_ecs_task_definition"
+  container_definitions    = "${data.template_file.ecs_task.rendered}"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "2048"
+  memory                   = "4096"
+  execution_role_arn       = "${aws_iam_role.ecs_execution_role.arn}"
+  task_role_arn            = "${aws_iam_role.ecs_execution_role.arn}"
+}
+
+resource "aws_ecs_task_definition" "ecs_task_definition_gitlab" {
+  count = "${var.scmgitlab}"
+  family                   = "${var.envPrefix}_ecs_task_definition_gitlab"
+  container_definitions    = "${data.template_file.ecs_task_gitlab.rendered}"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "2048"
+  memory                   = "4096"
+  execution_role_arn       = "${aws_iam_role.ecs_execution_role.arn}"
+  task_role_arn            = "${aws_iam_role.ecs_execution_role.arn}"
+}
+
+resource "aws_ecs_task_definition" "ecs_task_definition_codeq" {
+  count = "${var.codeq * var.scmgitlab * var.dockerizedJenkins}"
+  family                   = "${var.envPrefix}_ecs_task_definition_codeq"
+  container_definitions    = "${data.template_file.ecs_task_codeq.rendered}"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = "1024"
+  memory                   = "2048"
+  execution_role_arn       = "${aws_iam_role.ecs_execution_role.arn}"
+  task_role_arn            = "${aws_iam_role.ecs_execution_role.arn}"
+}
+
+resource "aws_alb_target_group" "alb_target_group" {
+  count = "${var.dockerizedJenkins}"
+  name     = "${var.envPrefix}-ecs-tg"
+  port     = 80
+  protocol = "HTTP"
+  vpc_id   = "${lookup(var.jenkinsservermap, "jenkins_vpc_id")}"
+  target_type = "ip"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  health_check {
+    path             = "/login"
+    matcher          = "200"
+    interval         = "60"
+    timeout          = "59"
+  }
+}
+
+resource "aws_alb_target_group" "alb_target_group_gitlab" {
+  count = "${var.scmgitlab}"
+  name     = "${var.envPrefix}-ecs-gitlab-tg"
+  port     = 80
+  protocol = "HTTP"
+  vpc_id   = "${lookup(var.jenkinsservermap, "jenkins_vpc_id")}"
+  target_type = "ip"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  health_check {
```

Comment:
Used this - -/health

Reply:
reverted the change since, seems, :  To access monitoring resources, the client IP needs to be included in a whitelist.

Reply:
Yes, and you can set the client IP healthcheck whitelist via env var, as described [here](https://docs.gitlab.com/ee/administration/monitoring/ip_whitelist.html). Does that not work here?

Reply:
Not sure. Since ECS fargate has dynamic IP (private and public) ranges. Also for health check, the registered target will be private IP. The URL has to be accessible for the registered targets health check.

Reply:
Fair enough, we can take it as a possible improvement for later on.